router/transformer/transformer_test.go

```diff
@@ -0,0 +1,232 @@
+				"XML":        map[string]interface{}{},
+			},
+			Files: map[string]interface{}{},
+		},
+	},
+	"ctx_cancel_dest": {
+		name:            "should throw timeout exception due to context getting cancelled immediately",
+		ExpectedStCode:  http.StatusInternalServerError,
+		ExpectedBody:    `Post "%s/v0/destinations/ctx_cancel_dest/proxy": context canceled`,
+		proxyStatusCode: http.StatusOK,
+		proxyResponse:   `{"output": {"status": 200, "message": "", "destinationResponse":"ctx_cancel_dest"}}`,
+		context: ProxyContextTc{
+			Cancel: true,
+		},
+		postParametersT: integrations.PostParametersT{
+			Type:          "REST",
+			URL:           "http://www.ctx_timeout_dest.domain.com",
+			RequestMethod: http.MethodPost,
+			QueryParams:   map[string]interface{}{},
+			Body: map[string]interface{}{
+				"JSON": map[string]interface{}{
+					"key_1": "val_1",
+					"key_2": "val_2",
+				},
+				"FORM":       map[string]interface{}{},
+				"JSON_ARRAY": map[string]interface{}{},
+				"XML":        map[string]interface{}{},
+			},
+			Files: map[string]interface{}{},
+		},
+	},
+}
+
+func Initialization() {
+	config.Load()
+	stats.Init()
+	stats.Setup()
+	rtTf.Init()
+	logger.Init()
+}
+
+func TestProxyRequest(t *testing.T) {
+	t.Setenv("RSERVER_HTTP_CLIENT_TIMEOUT", "1s")
+	Initialization()
+
+	srvMux := mux.NewRouter()
+	srvMux.HandleFunc("/v0/destinations/{destName}/proxy", proxyHandlerFunc)
+	srv := httptest.NewServer(srvMux)
+	defer srv.Close()
+
+	for destName, tc := range proxyResponseTcs {
+		t.Run(tc.name, func(t *testing.T) {
+			tr := rtTf.NewTransformer()
+			// Logic for executing test-cases not manipulating test-cases
+			if tc.rtTimeout.Milliseconds() > 0 {
+				tr.Setup(tc.rtTimeout)
+			} else {
+				// Just a default value
+				tr.Setup(2 * time.Millisecond)
+			}
+			// Logic to include context timing out
+			ctx := context.TODO()
+			var cancelFunc context.CancelFunc
+			if tc.context.Timeout.Milliseconds() > 0 {
+				ctx, cancelFunc = context.WithTimeout(context.TODO(), tc.context.Timeout)
+				defer cancelFunc()
+			} else if tc.context.Cancel {
+				ctx, cancelFunc = context.WithCancel(context.TODO())
+				cancelFunc()
+			}
+
+			reqParams := &rtTf.ProxyRequestParams{
+				ResponseData: tc.postParametersT,
+				DestName:     destName,
+				BaseUrl:      srv.URL,
+			}
+			stCd, resp := tr.ProxyRequest(ctx, reqParams)
+
+			assert.Equal(t, tc.ExpectedStCode, stCd)
+			if gjson.GetBytes([]byte(resp), "message").Raw != "" {
+				require.JSONEq(t, tc.ExpectedBody, resp)
+			} else if tc.ExpectedStCode == http.StatusGatewayTimeout {
+				expectedBodyStr := fmt.Sprintf(tc.ExpectedBody, srv.URL)
+				require.Equal(t, expectedBodyStr, resp)
+			}
+		})
+	}
+}
+
+// A kind of mock for transformer proxy endpoint in transformer
+func proxyHandlerFunc(w http.ResponseWriter, req *http.Request) {
+	vars := mux.Vars(req)
+	destName, ok := vars["destName"]
+	if !ok {
+		// Modify this case ?
+		panic(fmt.Errorf("Wrong url being sent"))
```

Comment:
Instead of panicking, can we use writer to return error response?
use return http.Error(...), same for all panics in this functions

Reply:
Done